godot-macros/src/class/data_models/rpc.rs

```diff
@@ -0,0 +1,150 @@
+ */
+
+use crate::class::FuncDefinition;
+use proc_macro2::{Ident, TokenStream};
+use quote::quote;
+
+/// Possible ways the user can specify RPC configuration.
+pub enum RpcInfo {
+    // Individual keys in the `rpc` attribute.
+    // Example: `#[rpc(any_peer, reliable, call_remote, channel = 3)]`
+    SeparatedArgs(RpcSeparatedArgs),
+    // `args` key in the `rpc` attribute.
+    // Example:
+    // const RPC_ARGS: RpcArgs = RpcArgs { mode: RpcMode::Authority, ..RpcArgs::default() };
+    // #[rpc(args = RPC_ARGS)]
+    Expression(TokenStream),
+}
+
+/// See [Godot documentation](https://docs.godotengine.org/en/stable/tutorials/networking/high_level_multiplayer.html#remote-procedure-calls)
+pub struct RpcSeparatedArgs {
+    pub mode: Option<RpcMode>,
+    pub transfer_mode: Option<TransferMode>,
+    pub call_local: Option<bool>,
+    pub transfer_channel: Option<u32>,
+}
+
+pub enum RpcMode {
+    Authority,
+    AnyPeer,
+}
+
+pub enum TransferMode {
+    Reliable,
+    Unreliable,
+    UnreliableOrdered,
+}
+
+pub fn make_rpc_registrations_fn(
+    class_name: &Ident,
+    funcs: &mut [FuncDefinition],
+) -> Option<TokenStream> {
+    let rpc_registrations = funcs
+        .iter_mut()
+        .filter_map(make_rpc_registration)
+        .collect::<Vec<TokenStream>>();
+
+    // This check is necessary because the class might not implement `WithBaseField` or `Inherits<Node>`,
+    //   which means `to_gd` wouldn't exist or the trait bounds on `RpcArgs::register` wouldn't be satisfied.
+    if rpc_registrations.is_empty() {
+        return None;
+    }
+
+    let result = quote! {
+        fn __register_rpcs(object: &mut dyn ::std::any::Any) {
+            use ::std::any::Any;
+            use ::godot::meta::RpcArgs;
+            use ::godot::classes::multiplayer_api::RpcMode;
+            use ::godot::classes::multiplayer_peer::TransferMode;
+            use ::godot::classes::Node;
+            use ::godot::obj::{WithBaseField, Gd};
+
+            let mut gd = object
+                .downcast_mut::<#class_name>()
+                .expect("bad type erasure")
+                .to_gd();
+
+            let node = gd.upcast_mut::<Node>();
+            #( #rpc_registrations )*
+        }
+    };
+
+    Some(result)
+}
+
+fn make_rpc_registration(func_def: &mut FuncDefinition) -> Option<TokenStream> {
+    let rpc_info = func_def.rpc_info.take()?;
+
+    let create_struct = match rpc_info {
+        RpcInfo::SeparatedArgs(RpcSeparatedArgs {
+            mode,
+            transfer_mode,
+            call_local,
+            transfer_channel,
+        }) => {
+            let override_mode = mode.map(|mode| {
+                let token = match mode {
+                    RpcMode::Authority => quote! { RpcMode::AUTHORITY },
+                    RpcMode::AnyPeer => quote! { RpcMode::ANY_PEER },
+                };
+
+                quote! { let args = RpcArgs { mode: #token, ..args }; }
```

Comment:
I did it that way because of your comment on the other issue:

![image](https://github.com/user-attachments/assets/62e1884c-7f35-441f-81be-e9f5b09febd1)

> Generated code by the macro -- one call for each key: